watertap/core/zero_order_costing.py

```diff
@@ -752,6 +806,72 @@ def cost_gac(blk):
         blk.config.flowsheet_costing_block.cost_flow(
             blk.unit_model.activated_carbon_demand[t0], "activated_carbon")
 
+    def cost_mabr(blk):
+        """
+        General method for costing membrane aerated biofilm reactor. Capital cost
+        is based on the cost of reactor and blower.
+
+        This method also registers the electricity demand as a costed flow.
+        """
+        t0 = blk.flowsheet().time.first()
+
+        # Get parameter dict from database
+        parameter_dict = \
+            blk.unit_model.config.database.get_unit_operation_parameters(
+                blk.unit_model._tech_type,
+                subtype=blk.unit_model.config.process_subtype)
+
+        # Get costing parameter sub-block for this technology
+        A, B, C, D = _get_tech_parameters(
+            blk,
+            parameter_dict,
+            blk.unit_model.config.process_subtype,
+            ["specific_removal",
+             "reactor_cost",
+             "specific_air_flow",
+             "blower_cost"])
+
+        # Add cost variable and constraint
+        blk.capital_cost = pyo.Var(
+            initialize=1,
+            units=blk.config.flowsheet_costing_block.base_currency,
+            bounds=(0, None),
+            doc="Capital cost of unit operation")
+
+        Q_treated = pyo.units.convert(blk.unit_model.properties_treated[t0].flow_vol,
+                              to_units=pyo.units.m ** 3 / pyo.units.hour)
+
+        ammonium_reacted = pyo.units.convert(
+            blk.unit_model.properties_treated[t0].conc_mass_comp["ammonium_as_nitrogen"],
+            to_units=pyo.units.kg / pyo.units.m ** 3)
+
+        SV_reactor = Q_treated*ammonium_reacted
+
+        ES_reactor = (1/pyo.units.convert(A, to_units=pyo.units.kg / pyo.units.m**2 / pyo.units.hour))*SV_reactor
+
+        DCC_reactor = pyo.units.convert(ES_reactor*B, to_units=blk.config.flowsheet_costing_block.base_currency)
+
+        ES_blower = C*pyo.units.convert(blk.unit_model.blower_size, to_units=pyo.units.m ** 2)
+
+        DCC_blower = pyo.units.convert(ES_blower*D, to_units=blk.config.flowsheet_costing_block.base_currency)
```

Comment:
As above, you can save some unit conversion operations by combining many of these. In the end, I think you should only need two unit conversions.